contracts/multisig-prover/src/query.rs

```diff
@@ -28,7 +34,10 @@ pub fn get_proof(deps: Deps, multisig_session_id: Uint64) -> StdResult<GetProofR
         MultisigState::Pending => ProofStatus::Pending,
         MultisigState::Completed { .. } => {
             let execute_data = batch
-                .encode_execute_data(multisig.quorum, multisig.signers)
+                .encode_execute_data(
+                    multisig.quorum,
+                    optimize_signers(multisig.quorum, multisig.signers),
```

Comment:
```suggestion
                    optimize_signers(multisig.signers, multisig.quorum),
```
makes more sense to me this way around